File: packages/dd-trace/test/llmobs/parity/scenarios/openai/azure-chat-completion.py
import os

from openai import AzureOpenAI

from common import finish, start


start("openai")
client = AzureOpenAI(
    api_key=os.environ["OPENAI_API_KEY"],
    api_version="2024-02-01",
    azure_endpoint=f'{os.environ["PROVIDER_BASE_URL"]}/azure',
)
client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[{"role": "user", "content": "Say parity"}],
    max_tokens=8,
    temperature=0,
)
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/openai/azure-chat-completion.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('openai')
  const OpenAI = loadVersionedModule('openai')
  const client = new OpenAI.AzureOpenAI({
    apiKey: process.env.OPENAI_API_KEY,
    apiVersion: '2024-02-01',
    endpoint: `${process.env.PROVIDER_BASE_URL}/azure`,
  })
  await client.chat.completions.create({
    model: 'gpt-4o-mini',
    messages: [{ role: 'user', content: 'Say parity' }],
    max_tokens: 8,
    temperature: 0,
  })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})
